oicd/viewsets.py

```diff
@@ -96,21 +101,27 @@ def callback(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
                             data[self.map_claim_to_model.get(k)] = v
 
                     if (
-                        user_model.objects.filter(username=data.get("username")).count()
+                        user_model.objects.filter(
+                            username__iexact=data.get("username")
+                        ).count()
                         == 0
                     ):
+                        if not data.get("first_name") and not data.get("last_name"):
+                            return Response(
+                                _("Missing required fields: family_name, given_name"),
+                                status=status.HTTP_400_BAD_REQUEST,
+                            )
+
+                        if not data.get("first_name"):
+                            data["first_name"] = data.get("last_name")
```

Comment:
Based on [the documentation](https://ona.io/static/docs/profiles.html#register-a-new-user), `first_name` is a requirement; `last_name` doesn't seem to be a requirement - like we can add a new user without a last name. Can we trigger an error if `first_name` hasn't been POSTed and continue processing if we only have `first_name` without `last_name`?

Reply:
Yes, this code currently does that(_Though a bit differently_). Currently it'll only raise an error if both fields are missing but will use the `last_name` as the first name if present.

_Doing this because in some cases a user may only have one name and that may be returned as the "last_name" by the authentication server_